src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -721,116 +722,29 @@ public AssetSpecification buildAssetFromSnapshotAccessRequest(
   }
 
   @VisibleForTesting
-  List<SnapshotBuilderTable> pullTables(SnapshotAccessRequestResponse snapshotRequestModel) {
-    var valueSets = snapshotRequestModel.getSnapshotSpecification().getValueSets();
-    var valueSetNames = valueSets.stream().map(SnapshotBuilderFeatureValueGroup::getName).toList();
-
-    Map<String, SnapshotBuilderTable> tableMap = populateManualTableMap();
-
-    Set<String> missing = new HashSet<>(valueSetNames);
-    missing.removeAll(tableMap.keySet());
+  List<SnapshotBuilderTable> pullTables(SnapshotAccessRequestResponse snapshotAccessRequest) {
+    List<String> includedTableNames =
+        snapshotAccessRequest.getSnapshotSpecification().getOutputTables().stream()
+            .map(SnapshotBuilderOutputTable::getName)
+            .toList();
+    List<SnapshotBuilderDatasetConceptSet> allTables =
+        snapshotBuilderSettingsDao
+            .getBySnapshotId(snapshotAccessRequest.getSourceSnapshotId())
+            .getDatasetConceptSets();
+
+    Set<String> missing = new HashSet<>(includedTableNames);
+    allTables.stream()
+        .map(SnapshotBuilderDatasetConceptSet::getName)
+        .toList()
+        .forEach(missing::remove);
```

Comment:
There is no need to use `.toList()` here since you can use `forEach()` on a stream.
```suggestion
        .forEach(missing::remove);
```